sklearn/ensemble/bagging.py

```diff
@@ -454,14 +483,14 @@ def _validate_estimator(self):
         super(BaggingClassifier, self)._validate_estimator(
             default=DecisionTreeClassifier())
 
-    def _set_oob_score(self, X, y):
+    def _set_oob_score(self, X, y, n_more_estimators):
         n_classes_ = self.n_classes_
         classes_ = self.classes_
         n_samples = y.shape[0]
 
         predictions = np.zeros((n_samples, n_classes_))
 
-        for estimator, samples, features in zip(self.estimators_,
+        for estimator, samples, features in zip(self.estimators_[-n_more_estimators:],
                                                 self.estimators_samples_,
                                                 self.estimators_features_):
```

Comment:
`self.estimators_[-n_more_estimators:], self.estimators_samples_` and `self.estimators_features_` are all of the same size. 
- `self.estimators_[-n_more_estimators:]` contains the estimators trained during the last batch.
- `self.estimators_samples_` and `self.estimators_features_` contains the samples (resp. features) indices used during the last batch. 
Then the oob score is computed from the last batch only, as currently done in the forest as well. 

Why do think this is a bug @arjoly ?

I agree however that it may be nice to accumulate oob scores over the batches, but this would require a different logic from what we have right now (here and in forests).